lib/services/location_search.dart

```diff
@@ -1,58 +1,115 @@
   @override
   Widget buildSuggestions(BuildContext context) {
-    return ListView(
-      children: <Widget>[
-        Consumer<MapData>(
-          ///Wrapped in Consumer, listening to Provider **ConcreteObserver**
-          builder: (context, mapData, child) {
-            return ListTile(
-              leading: Icon(Icons.location_city),
-              title: Text("SGW Campus, Montreal"),
-              subtitle: Text("Quebec, Canada"),
-              onTap: () {
-                Navigator.of(context).pop();
-                mapData.controllerDestination.text = "SGW, Montreal";
-                mapData.controllerStarting.text = "Current Location";
-                mapData.changeStart(mapData.getCurrentLocation);
-                mapData.changeCampus('sgw');
-                mapData.changeEnd(sgw);
-                mapData.changeMode("driving");
-                Navigator.pushNamed(context, '/directions');
-              },
-            );
-          },
-        ),
-        Consumer<MapData>(
-          ///Wrapped in Consumer, listening to Provider **ConcreteObserver**
-          builder: (context, mapData, child) {
-            return ListTile(
-              leading: Icon(Icons.location_city),
-              title: Text("Loyola Campus, Montreal"),
-              subtitle: Text("Quebec, Canada"),
-              onTap: () {
-                Navigator.of(context).pop();
-                mapData.controllerDestination.text = "Loyola Campus, Montreal";
-                mapData.controllerStarting.text = "Current Location";
-                mapData.changeStart(mapData.getCurrentLocation);
-                mapData.changeCampus('loyola');
-                mapData.changeEnd(loyola);
-                mapData.changeMode("driving");
-                Navigator.pushNamed(context, '/directions');
-              },
-            );
-          },
-        ),
-      ],
+    getClassrooms();
+
+    final suggestionList = query.isEmpty
+        ? recentRooms
+        : classrooms.where((p) => p.contains(query.toUpperCase())).toList();
+    return ListView.builder(
+      itemBuilder: (context, index) => ListTile(
+        onTap: () async {
+          Provider.of<MapData>(context, listen: false)
+              .controllerDestination
+              .text = "SGW Campus, Montreal";
+          Provider.of<MapData>(context, listen: false).controllerStarting.text =
+              "Current Location";
+          Navigator.of(context).pop();
+          Provider.of<MapData>(context, listen: false)
+              .controllerDestination
+              .text = "SGW, Montreal";
+          Provider.of<MapData>(context, listen: false).controllerStarting.text =
+              "Current Location";
+          Provider.of<MapData>(context, listen: false).changeStart(
+              Provider.of<MapData>(context, listen: false).getCurrentLocation);
+          switch (suggestionList[index][0].toString()) {
```

Comment:
should we extract this algorithm into its own method?

Reply:
check https://github.com/MichelRahme/ConcordiaNavigation/blob/junk/98-directions-list/lib/providers/itinerary_data.dart i started doing that a bit there but nothing came of it